### PR TITLE
Allow for GtkApplication to be subclasses

### PR DESCRIPTION
**Pull request by alatiera**
This expands further upon the gio MR that does the same
for GApplictiona.

https://github.com/gtk-rs/gio/pull/209

### CODE REVIEW

File: src/subclass/mod.rs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// Copyright 2019, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+pub mod application;
```
Comment (alatiera) [inline src/subclass/mod.rs]: this nees a prelude

File: src/subclass/application.rs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    fn window_added(&self, application: &Application, window: &Window) {
+        self.parent_window_added(application, window)
+    }
+
+    fn window_removed(&self, application: &Application, window: &Window) {
+        self.parent_window_removed(application, window)
+    }
+}
+
+pub trait ApplicationImplExt {
```
Comment (sdroege) [inline src/subclass/application.rs]: All the traits should probably get `Gtk` as prefix

**Comment by sdroege** [comment]
Looks good otherwise :)

File: src/subclass/application.rs (hunk 2 above)
Comment (alatiera) [reply src/subclass/application.rs]: Done :)

File: Gir.toml [hunk 3]
```diff
@@ -426,6 +426,7 @@ status = "generate"
 name = "Gtk.Application"
 status = "generate"
 trait_name = "GtkApplicationExt"
+subclassing = true
```
Comment (EPashkin) [inline Gir.toml]: @sdroege This options now not used, can we delete it later?

Comment (sdroege) [reply Gir.toml]: I guess we can delete it now already :)

Comment (EPashkin) [reply Gir.toml]: Ok, I will do it.
@alatiera Please remove this line

Comment (alatiera) [reply Gir.toml]: done :)

**Comment by sdroege** [comment]
@GuillaumeGomez good to go then :)

**Comment by GuillaumeGomez** [comment]
Thanks @alatiera!

**Pull request merged**